### DISCUSSION TITLE
expanding the environment variables ....

### DISCUSSION THREAD

**Discussion by jglezplatas** [fortran-lang/fpm#683] [Ideas]
Is possible to expand the content of the environment variables to use with options as -I or -L in --flag and so on?
Example:
--flag " -L$LIB -I$INCLUDE"

Javier

**Comment by zoziha** [comment]
### For `-L$path`
There's already a PR (see #671) on this, but maybe this will lead to fpm being able to release **binary-only** fpm packages, **we need to be careful about it.**

(In my opinion: )
For package managers, `-Lpath` maybe not strictly required.
For build systems, `-Lpath` is necessary.

Currently, if you are using gfortran, you can support `-Lpath` by setting the `LIBRARY_PATH` environment variable.

### For `-I$path`
Currently fpm already **partially** supports `-I` for **relative paths** through `include-dir` in `fpm.toml` (see [include-directory](https://fpm.fortran-lang.org/en/spec/manifest.html#include-directory)).

### For `-l$lib`
Currently fpm already partially supports `-l` for relative paths through `link` in `fpm.toml` (see [link-external-libraries](https://fpm.fortran-lang.org/en/spec/manifest.html#link-external-libraries)).

**Reply by jglezplatas** [reply]
It is not exactly what I say.
My comment is to have the possibility to expand the environment variables when you need to describe any path directory in the options.
Probably you are thinking that all your dependencies are located in correct places but sometimes you are using particular libraries in particular directories that highly depend on the user choice.

And my last question is how to use this possibility:
In windows, should I use %varname% while in linux/macos $varname?

**Comment by jglezplatas** [comment]
Upsss!
I have tested using in a response file the possibility to use an environment variable and it works!
For example:
option --flag "-I %WINTER%\lib.i64"

That's great!
